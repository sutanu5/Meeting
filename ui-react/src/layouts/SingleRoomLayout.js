import React, { Component } from 'react';

import Display from '../components/single-room/Display';
import NotFound from '../components/global/NotFound';

class SingleRoomLayout extends Component {
  constructor(props) {
    super(props);
    this.state = {
      roomAlias: props.match.params.name
    }
  }

  render () {
    const { roomAlias } = this.state;
    return (
      <div id="single-room__wrap">
        { roomAlias ?
          <Display alias={roomAlias} />
        :
          <div id="error-wrap">
            <NotFound />
          </div>
        }
      </div>
    )
  }
}

export default SingleRoomLayout;
