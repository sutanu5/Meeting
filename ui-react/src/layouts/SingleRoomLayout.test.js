import SingleRoomLayout from './SingleRoomLayout';


describe('SingleRoomLayout Component', () => {
  let props, wrapper;;
  beforeEach(() => {
    props = {
      match: {
        params: {
          name: 'test-room'
        }
      }
    };

    wrapper = shallow(<SingleRoomLayout {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Display component if roomAlias exists', () => {
    const component = wrapper.find('Display').exists();
    expect(component).toBeTruthy();
  });

  it('renders Display component props correctly', () => {
    const alias = wrapper.find('Display').props().alias;
    expect(alias).toEqual(props.match.params.name);
  })

  it('renders NotFound if roomAlias does not exists', () => {
    wrapper.setState({ roomAlias: '' });

    const component = wrapper.find('NotFound').exists();
    expect(component).toBeTruthy();
  });
});