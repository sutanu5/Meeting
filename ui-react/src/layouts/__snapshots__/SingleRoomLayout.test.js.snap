// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRoomLayout Component renders correctly 1`] = `
<div
  id="single-room__wrap"
>
  <Display
    alias="test-room"
  />
</div>
`;
